### PR TITLE
Reduce alert events sent to the cloud.

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

The purpose of this PR is to reduce the number of alert events sent to the cloud. Right now the agent does not apply any specific rules on which alert events to send, thus in some situations (mostly during restarts) will send events that are of no use to the cloud experience.

This PR tries to skip consecutive same alert events if:

* The alert configuration has not changed
* There is a less than 15 minutes difference between a REMOVED event and a new WARNING or CRITICAL status. If the new status is CLEAR we skip this check.

It will also handle better the case of an agent claimed while running. On the first start streaming request for batch_id 1 and start sequence 1, it will try to queue the existing alerts to the cloud. The relevant query there also takes into account the alert events having a config hash (https://github.com/netdata/netdata/issues/12369).

The REMOVED events created on chart obsoletion will also be queued immediately, skipping the filter check above.

##### Test Plan

Test will be done together with the set team to make sure we don't miss anything important, and to cover extra cases. It will be converted to proper PR after this.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  

Should reduce the amount of traffic between agent and cloud, and also the burden on the cloud.
</details>

### CODE REVIEW

**Comment by MrZammler** [comment]
Just a ping, converted from Draft to Ready for review. Also, we conducted tests with @dimko (thanks!) on staging.

File: database/sqlite/sqlite_aclk_alert.c [hunk 1]
```diff
@@ -8,9 +8,120 @@
...
+        error_report("Failed to prepare statement when trying to find removed gap.");
+        return 0;
+    }
+
+    rc = sqlite3_step(res);
+    if (likely(rc == SQLITE_ROW)) {
+        when = (time_t) sqlite3_column_int64(res, 0);
+    }
+
+    rc = sqlite3_reset(res);
```
Comment (stelfrag) [inline database/sqlite/sqlite_aclk_alert.c:32]: This should be finalize

File: database/sqlite/sqlite_aclk_alert.c [hunk 2]
```diff
@@ -8,9 +8,120 @@
...
+            send = 1;
+            goto done;
+        } else {
+            send = 0;
+            goto done;
+        }
+    }
+     
+done:
+    rc = sqlite3_reset(res);
```
Comment (stelfrag) [inline database/sqlite/sqlite_aclk_alert.c:115]: This should be finalize

File: database/sqlite/sqlite_aclk_alert.c [hunk 3]
```diff
@@ -8,9 +8,120 @@
...
+    snprintfz(sql,ACLK_SYNC_QUERY_SIZE-1, "select hl.new_status, hl.config_hash_id, hl.unique_id from health_log_%s hl, aclk_alert_%s aa \
+                         where hl.unique_id = aa.alert_unique_id        \
+                         and hl.alarm_id = %u and hl.unique_id <> %u \
+                         order by alarm_event_id desc LIMIT 1;", uuid_str, uuid_str, ae->alarm_id, ae->unique_id);
+
+    rc = sqlite3_prepare_v2(db_meta, sql, -1, &res, 0);
+    if (rc != SQLITE_OK) {
+        error_report("Failed to prepare statement when trying to filter alert events.");
+        send = 1;
+        goto done;
```
Comment (stelfrag) [inline database/sqlite/sqlite_aclk_alert.c:70]: This is invalid, done will do reset (should be finalize) but prepare has failed

File: database/sqlite/sqlite_aclk_alert.c (hunk 1 above)
Comment (MrZammler) [reply database/sqlite/sqlite_aclk_alert.c]: Embarrassing...

File: database/sqlite/sqlite_aclk_alert.c (hunk 3 above)
Comment (MrZammler) [reply database/sqlite/sqlite_aclk_alert.c]: Should be ok now, thanks...

**Review by stelfrag — APPROVED** [review]

**Comment by MrZammler** [comment]
Thank you!

**Pull request merged**